src/backend/gl/src/command.rs

```diff
@@ -1274,10 +1294,14 @@ impl command::RawCommandBuffer<Backend> for RawCommandBuffer {
         &mut self,
         _layout: &n::PipelineLayout,
         _stages: pso::ShaderStageFlags,
-        _offset: u32,
-        _constants: &[u32],
+        offset: u32,
+        constants: &[u32],
     ) {
-        unimplemented!()
+        let constants = self.add(constants);
+        self.push_cmd(Command::BindUniform(
```

Comment:
Is this valid? How does spirv-cross translate push constants to GLSL? I would assume a uniform struct, so I'm wondering a bit if filling it with a raw u32 array is allowed

Reply:
spirv-cross translates push constants to a uniform struct, so my idea is that with 32-bit words push constants in this fashion would be similar to a memcpy. Is there a more elegant way to do this @msiglreith?

Reply:
Could you print an example of the spirv-cross output?
I would assume we would need to do some sort of reflection here at least to correctly fill all the fields

Reply:
```rust
#version 450

struct PushConsts
{
    mat4 view_proj;
};

uniform PushConsts push;

layout(location = 2) in vec4 model_col1;
layout(location = 3) in vec4 model_col2;
layout(location = 4) in vec4 model_col3;
layout(location = 5) in vec4 model_col4;
layout(location = 0) in vec3 position;
layout(location = 1) out vec2 frag_uv;
layout(location = 1) in vec2 vert_uv;

void main()
{
    mat4 model = mat4(vec4(model_col1), vec4(model_col2), vec4(model_col3), vec4(model_col4));
    gl_Position = (push.view_proj * model) * vec4(position, 1.0);
    frag_uv = vert_uv;
}
```

Reply:
IIRC gl assignes different locations for fields of uniforms so each field needs to be manually assigned -> reflection required

Reply:
Hey @msiglreith check out my new push constant to uniform translations.